src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -434,9 +425,8 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         path_str: &'a str,
         module_id: DefId,
     ) -> Result<Res, ResolutionFailure<'a>> {
-        let cx = self.cx;
         let path = ast::Path::from_ident(Ident::from_str(path_str));
-        cx.enter_resolver(|resolver| {
+        self.cx.enter_resolver(|resolver| {
```

Comment:
This change isn't strictly necessary, but it avoids having to write `let cx = &self.cx;`, which looks very silly since cx is a mutable reference.

Reply:
> This change isn't strictly necessary, but it avoids having to write `let cx = &self.cx;`, which looks very silly since cx is a mutable reference.

Wait... how can it be a mutable reference? Do you mean an `&&mut ...`? I thought you couldn't mutate through those...

Reply:
You can't, everything that uses it only needs &DocContext (just like it did before). Essentially it's reborrowing the &&mut T as &&T at every use and then using the fact that & is Copy to get a &T.

Reply:
Oh... I see. 👍